### ISSUE TITLE
Default pagination limit of Snippets is too high (99999)

### ISSUE THREAD

**Issue by effixe** [sulu/sulu#4130]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.6.*

#### Actual Behavior

When loading a snippet list in a Block the default pagination is set to 99999.
In my case I have quite a lot of snippets and when I have to add one to the Block it returns a timeout.

#### Expected Behavior

When loading a snippet list in a Block the default pagination is set to 50. No more timeout and no more very long scrolling.

#### Steps to Reproduce

Just add 1000 snippets and try to load the list in a Block.

#### Possible Solutions

Just set the limit at 50 in sulu/src/Sulu/Bundle/SnippetBundle/Ressources/public/js/components/snippet-content/main.js line 109

**Comment by danrot** [comment]
Are you talking about the assignment of a snippet to e.g. a page? Just setting it to a lower value probably doesn't cut it, since then the content manager doesn't see all the assigned snippets...

**Comment by danrot** [comment]
Ok, and don't you get performance problems on the website when adding so many snippets to the page? Since this content type also immediately loads all the snippets it seems inevitable. The content type is simply not build for assigning such a huge amount... (it's also not comfortable to use). Wouldn't make using a [SmartContent](http://docs.sulu.io/en/latest/reference/content-types/smart_content.html) with a tag or category filter more sense? It would also allow you to paginate the results.

**Comment by FXBPinelli** [comment]
Actually I may have misspoke myself.

The problem is in the backoffice when I try to load the list of snippets (via a `type="snippet"` property). I don't display a thouthand snippets in my twig.

**Comment by danrot** [comment]
I have understood correctly. But when you add a thousand snippets to the snippet selection, then all of the snippets are also loaded on the website, no matter if you display them or not. That's why I am asking if you don't have performance problems on the website. And why is the SmartContent no solution for you?

**Comment by danrot** [comment]
In Sulu 2.0 we use the ID to make this load... But this will result in the same problem...

**Comment by danrot** [comment]
We've discussed this today, and we came to the conclusion that one issue in the administration interface that does not occur on the website is that there might be a limit on how many IDs we can add to the URL as query parameter. Therefore it might make sense to add a "Load more" button and only 20 or 50 entities at once.

However, this currently does not have a very high priority for us, since I've already mentioned that this is more like an edge case.
